app/models/concerns/galtersufia/generic_file/invenio_resource_type_mappings.rb

```diff
@@ -4,177 +4,76 @@ module InvenioResourceTypeMappings
-        "Conference Proceeding": "Conference Objects",
-        "Congress": "Conference Objects",
-        "Meeting Abstract": "Conference Objects",
-        "Dataset": "Dataset",
-        "Architectural Drawing": "Images",
-        "Chart": "Images",
-        "Drawing": "Images",
-        "Map": "Images",
-        "Photograph": "Images",
-        "Pictorial Work": "Images",
-        "Portrait": "Images",
-        "Examination Questions": "Learning Objects",
-        "Lecture": "Learning Objects",
-        "Lecture Notes": "Learning Objects",
-        # "Lesson Plan": "Learning Objects",
-        "Presentation": "Learning Objects",
-        # "Problems and Exercises": "Learning Objects",
-        # "Syllabus": "Learning Objects",
-        "Animation": "Multimedia",
-        # "Database": "Multimedia",
-        "Postcard": "Multimedia",
-        "Social Media Posting": "Multimedia",
-        "Software": "Multimedia",
-        "Sound (Recording)": "Multimedia",
-        "Video": "Multimedia",
-        # "Website": "Multimedia",
-        # "Interview": "Other",
-        # "Laboratory Manual": "Other",
-        "News": "Other",
-        "Other": "Other",
-        "Table": "Other",
-        "Journal": "Periodicals",
-        "Magazine": "Periodicals",
-        "Newsletter": "Periodicals",
-        # "Newspaper": "Periodicals",
-        "Case Report": "Study Documentation",
-        # "Clinical Study": "Study Documentation",
-        "Clinical Trial": "Study Documentation",
-        # "Comparative Study": "Study Documentation",
-        "Data Management Plan": "Study Documentation",
-        "Evaluation Study": "Study Documentation",
-        # "Measurements and Tests": "Study Documentation",
-        # "Research Proposal": "Study Documentation",
-        "Statistics": "Study Documentation",
-        "Study Design": "Study Documentation",
-        "Abstract": "Text Resources",
-        "Address (Speech)": "Text Resources",
-        "Bibliography": "Text Resources",
-        "Biobibliography": "Text Resources",
-        # "Comment": "Text Resources",
-        "Correspondence": "Text Resources",
-        "Fictional Work": "Text Resources",
-        "Form": "Text Resources",
-        "Guideline": "Text Resources",
-        "Letter": "Text Resources",
-        "Manuscript": "Text Resources",
-        # "Patent": "Text Resources",
-        # "Patient Education Handout": "Text Resources",
-        "Personal Narrative": "Text Resources",
-        "Poetry": "Text Resources",
-        "Preprint": "Text Resources",
-        "Program": "Text Resources",
-        "Report": "Text Resources",
-        "Resource Guide": "Text Resources",
-        "Technical Documentation": "Text Resources",
-        # "Working Paper": "Text Resources",
-        "Bachelor Thesis": "Theses and Dissertations",
-        "Doctoral Thesis": "Theses and Dissertations",
-        "Masters Thesis": "Theses and Dissertations"
+      DH_IRDM_RESOURCE_TYPES = {
```

Comment:
Thanks for clearing this up and apologies for screwing it up so much the first time. I like the way you laid out the mapping here. Once I saw how you were using this it made a lot more sense to me why you took this approach.

Reply:
Nothing was screwed up to me. I thought you did the mapping the right way the first time. As for changing the mapping, I wanted to keep the original, but I went with what was clear to me immediately. Looking back it may have turned out better due to at least one record not having a subtype.